src/api/subscriptions.js

```diff
@@ -317,6 +317,72 @@ export const SubscriptionFlows = {
  */
 export let Config;
 
+/**
+ * Internal params for GetEntitlements requests.
+ * @typedef {{
+ *   encryption: (!GetEntitlementsEncryptionParams|undefined),
+ *   metering: (!GetEntitlementsMeteringParamsInternal|undefined),
+ * }}
+ */
+export let GetEntitlementsParamsInternal;
+
+/**
+ * Encryption params for GetEntitlements requests.
+ * @typedef {{
+ *   encryptedDocumentKey: string,
+ * }}
+ */
+export let GetEntitlementsEncryptionParams;
+
+/**
+ * Internal metering params for GetEntitlements requests.
+ * @typedef {{
+ *   clientTypes: !Array<number>,
+ *   owner: string,
+ *   state: {
+ *     id: string,
+ *     attributes: !Array<{
+ *       name: string,
+ *       timestamp: number,
+ *     }>,
+ *   },
+ *   resource: {
+ *     hashedCanonicalUrl: string,
+ *   },
+ * }}
+ */
+export let GetEntitlementsMeteringParamsInternal;
+
+/**
+ * External params for GetEntitlements requests.
+ * @typedef {{
+ *   encryption: (!GetEntitlementsEncryptionParams|undefined),
+ *   metering: (!GetEntitlementsMeteringParamsExternal|undefined),
+ * }}
+ */
+export let GetEntitlementsParamsExternal;
+
+/**
+ * External metering params for GetEntitlements requests.
+ * @typedef {{
+ *   clientTypes: !Array<number>,
+ *   owner: string,
+ *   state: {
+ *     id: string,
+ *     standardAttributes: !Object<string, {
+ *       timestamp: number,
+ *     }>,
+ *     customAttributes: !Object<string, {
+ *       timestamp: number,
+ *     }>,
+ *   },
+ *   resource: {
+ *     hashedCanonicalUrl: string,
+ *   },
+ * }}
+ */
+export let GetEntitlementsMeteringParamsExternal;
```

Comment:
I'd rather avoid having the word "External" in a public facing typedef.  From their perspective it is just GetEntitlementsMeteringParams.  Perhaps we can just have Internal for ours or come up with a better name so we don't need this descritpor?

Reply:
I would expect publishers would start by checking out the `publisher.js` example and/or a markdown file like `entitlements-flow.md`. Publishers won't need to reference this typedef in their code either. If they arrive at this typedef, it's because they were digging in the source code, at which point I think a term like `External` would be fairly descriptive. Please feel free to suggest clearer alternatives though?